File: core/acp-core/src/main/java/pers/acp/core/task/timer/ruletype/ExecuteType.java
package pers.acp.core.task.timer.ruletype;

import java.util.HashMap;
import java.util.Map;

import pers.acp.core.exceptions.EnumValueUndefinedException;
import pers.acp.core.interfaces.IEnumValue;

/**
 * 执行类型
 *
 * @author zhangbin
 */
public enum ExecuteType implements IEnumValue {

    WeekDay("WeekDay", 0),

    Weekend("Weekend", 1),

    All("All", 2);

    private String name;

    private Integer value;

    private static Map<Integer, ExecuteType> map;

    private static Map<String, ExecuteType> nameMap;

    static {
        map = new HashMap<>();
        nameMap = new HashMap<>();
        for (ExecuteType type : values()) {
            map.put(type.getValue(), type);
            nameMap.put(type.getName(), type);
        }
    }

    ExecuteType(String name, Integer value) {
        this.name = name.toLowerCase();
        this.value = value;
    }

    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public Integer getValue() {
        return this.value;
    }

    @Override
    public Boolean equals(Integer value) {
        return this.value.equals(value);
    }

    public static ExecuteType getEnum(Integer value) throws EnumValueUndefinedException {
        if (map.containsKey(value)) {
            return map.get(value);
        }
        throw new EnumValueUndefinedException(ExecuteType.class, value);
    }

    public static ExecuteType getEnum(String name) throws EnumValueUndefinedException {
        if (nameMap.containsKey(name.toLowerCase())) {
            return nameMap.get(name.toLowerCase());
        }
        throw new EnumValueUndefinedException(ExecuteType.class, name);
    }

}

File: core/acp-core/src/main/kotlin/pers/acp/core/task/timer/rule/ExecuteType.kt
package pers.acp.core.task.timer.rule

import pers.acp.core.exceptions.EnumValueUndefinedException
import java.util.HashMap

/**
 * @author zhang by 10/07/2019
 * @since JDK 11
 */
enum class ExecuteType {

    WeekDay,
    Weekend,
    All;

    companion object {

        private var nameMap: MutableMap<String, ExecuteType> = HashMap()

        init {
            for (type in values()) {
                nameMap[type.name.toUpperCase()] = type
            }
        }

        @Throws(EnumValueUndefinedException::class)
        fun getEnum(name: String): ExecuteType {
            if (nameMap.containsKey(name.toLowerCase())) {
                return nameMap.getValue(name.toLowerCase())
            }
            throw EnumValueUndefinedException(ExecuteType::class.java, name)
        }
    }

}